### DISCUSSION TITLE
Support for VIM4 machine type?

### DISCUSSION THREAD

**Discussion by mike-at-dusty** [superna9999/meta-meson#138] [Q&A]
Is the [VIM4](https://www.khadas.com/vim4) with the A311D2 being looked at as a supported machine type?

I can help out with it, but I was wondering if someone else had already done much of the work.

I'm currently building VIM3 images to understand what the delta might be between it and one that would support the VIM4.

I suspect the mainline kernel may not yet support the A311D2, but my hope is the [Khadas kernel for the VIM4](https://github.com/khadas/linux/commits/khadas-vims-5.4.y-yocto-1) should be able to be used.  Probably the same for Uboot.

**Comment by superna9999** [comment]
The question has already been asked at https://github.com/superna9999/meta-meson/issues/134

TL;DR there's no current ongoing effort so far. Amlogic is pushing changes for the S4 SoCs that could be used for the T7 SoC (A311D2).

This layer is designed only for upstream Linux & U-Boot, if you can provide an upstream based set of patches, I'll  be happy to integrate them.

You may ask Khadas if there's plans to start upstream support at some point,
